Tests/DatadogTests/Datadog/RUMMonitorTests.swift

```diff
@@ -330,6 +330,37 @@ class RUMMonitorTests: XCTestCase {
             }
     }
 
+    func testStartingView_thenTappingButton_thenTappingAnotherButton() throws {
```

Comment:
shouldn't this case be in `ViewScopeTests` as the fix done in `ViewScope`?

Reply:
We have two levels of RUM unit test:
* small tests - defined per scope (e.g. `RUMViewScopeTests`, `RUMUserActionScopeTests`),
* big tests  - defined in `RUMMonitorTests`.

Small tests use the scope instance directly, pass the `RUMCommand` for processing and expect on the scope state + assert produced events.

Big tests use the `RUMMonitor` public interface and assert uploaded events.

As big tests are much wider (they cover more code), they should be considered as more valuable. All the behaviours expressed in big tests can be also written as small tests, but this will produce longer testing code, more fragile to the implementation changes (as small test are closer to the implementation).

I decided to add the big test for this bug, as the bug was reported as user scenario (tap one button, wait, tap another button). I could move this test to the `RUMUserActionScopeTests` (rather than `RUMViewScopeTests` as we want to test the behaviour of UA Scope), but it won't be as widely covering as here. It will only assert that _"when expired UA scope receives the `RUMRegisterUserActionCommand`, it sends the `RUMAction` event and gets closed"_. Or I can add one more small test in the UA scope. WDYT @buranmert ?

Reply:
so long as we have this test case, it doesn't matter much to me where :)

i was asking bc to my understanding the bug&fix is about the way `ViewScope` manages `UserActionScope`, so `ViewScope` is the unit to test here. if i understand well, you see `RUMMonitorTests` like user scenarios 👌